src/services/repo/useEncodeString.tsx

```diff
@@ -0,0 +1,58 @@
+import { useMutation } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+
+import { useAddNotification } from 'services/toastNotification'
+import Api from 'shared/api'
+
+const query = `
+  mutation EncodeSecretString($repoName: String!, $value: String!) {
+    encodeSecretString(input: { repoName: $repoName, value: $value }) {
+      value
+      error {
+        ... on ValidationError {
+          __typename
+          message
+        }
+        ... on UnauthenticatedError {
+          __typename
+          message
+        }
+      }
+    }
+  }
+`
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export const useEncodeString = () => {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const addToast = useAddNotification()
+  return useMutation({
+    mutationFn: (value: string) => {
+      return Api.graphqlMutation({
+        provider,
+        query,
+        variables: {
+          owner,
+          repoName: repo,
+          value,
+        },
+        mutationPath: 'EncodeSecretString',
+      })
+    },
+    onSuccess: ({ data }) => {
+      const error = data?.encodeSecretString?.error
```

Comment:
Do you think it's worth validating these responses with zod schemas?

Reply:
I could see either way - curious your thoughts though. Not sure what we do for other mutations.